src/pages/api/user/get_balance.ts

```diff
@@ -0,0 +1,52 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "@/lib/server/prisma";
+import { verifyAuthToken } from "@/lib/server/auth";
+import { getClaveInviteLink, getUserBuidlBalance } from "@/lib/server/clave";
+import { ClaveInfo } from "@/lib/client/clave";
+import { ErrorResponse } from "@/types";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ClaveInfo | ErrorResponse>
+) {
+  try {
+    if (req.method !== "GET") {
+      res.setHeader("Allow", ["GET"]);
+      return res.status(405).end(`Method ${req.method} Not Allowed`);
+    }
+
+    // Extract the token from the query parameters
+    const token = req.query.token as string;
+    if (!token) {
+      return res.status(401).json({ error: "Authorization token is missing" });
+    }
+
+    const userId = await verifyAuthToken(token);
+    if (!userId) {
+      return res.status(401).json({ error: "Invalid token" });
+    }
+
+    const user = await prisma.user.findUnique({
+      where: { id: userId },
+    });
+    if (!user) {
+      return res.status(400).json({ error: "Invalid user id" });
+    }
+
+    const buidlBalance = await getUserBuidlBalance(user.id.toString());
+    const claveWalletAddress = user.claveWallet ? user.claveWallet : undefined;
+    const claveInviteLink = await getClaveInviteLink(
```

Comment:
generates a new invite link every time we check balance? seems incorrect

Reply:
it is incorrect. fixed